modules/components/src/AdvancedFacetView/FacetView.js

```diff
@@ -1,46 +1,106 @@
+import { css } from 'emotion';
+import { Subject } from 'rxjs';
+import AggsWrapper from '../Aggs/AggsWrapper';
+import aggComponents from './aggComponents';
+import TextHighlight from '../TextHighlight';
 
 const serializeToDomId = path => path.split('.').join('__');
 
+const flattenDisplayTreeData = displayTreeData => {
+  return displayTreeData.reduce(
+    (acc, node) => [
+      ...acc,
+      ...(node.children ? flattenDisplayTreeData(node.children) : [node]),
+    ],
+    [],
+  );
+};
+
 export default class FacetView extends React.Component {
-  scrollToPath = path => {
+  scrollToPath = ({ path, behavior = 'smooth', block = 'start' }) => {
     const targetElementId = serializeToDomId(path);
     const targetElement = this.root.querySelector(`#${targetElementId}`);
     if (targetElement) {
-      targetElement.scrollIntoView({ behavior: 'smooth', block: 'start' });
+      targetElement.scrollIntoView({ behavior, block });
     }
   };
+  $updated = new Subject();
+  componentDidUpdate() {
+    this.$updated.next();
+  }
   render() {
     const {
       aggregations,
       displayTreeData,
       onValueChange,
-      sqon = {},
+      sqon = null,
       constructEntryId,
       searchboxSelectionObservable,
       valueCharacterLimit,
+      extendedMapping,
+      searchString,
     } = this.props;
     return (
       <div className="facetView" ref={el => (this.root = el)}>
-        {displayTreeData.map(node => {
-          return (
-            <FacetViewNode
-              valueCharacterLimit={valueCharacterLimit}
-              searchboxSelectionObservable={searchboxSelectionObservable}
-              constructEntryId={constructEntryId}
-              depth={0}
-              sqon={sqon}
-              key={node.path}
-              path={node.path}
-              aggregations={aggregations}
-              onValueChange={({ value, path, esType, aggType }) =>
-                onValueChange({ value, path, esType, aggType })
-              }
-              {...node}
-            />
+        {flattenDisplayTreeData(displayTreeData).map(({ title, path }) => {
+          const metaData = extendedMapping.find(({ field }) => field === path);
+          const { type } = metaData || {};
+          const paths = path
+            .split('.')
+            .reduce(
+              (acc, node, i, paths) => [
+                ...acc,
+                [...paths.slice(0, i), node].join('.'),
+              ],
+              [],
+            );
+          const pathDisplayNames = paths.map(
+            path =>
+              extendedMapping.find(({ field }) => field === path)?.displayName,
           );
+          const agg = aggregations[path];
+          return aggComponents[type]?.({
+            ...agg,
+            key: path,
+            field: path,
+            onValueChange: ({ sqon }) => {
+              onValueChange({ sqon });
+              const updateSubs = this.$updated.subscribe(() => {
```

Comment:
Is this to get the scroll to happen after update? could you just setstate with the path and check if it changed in componentDidUpdate?

Reply:
yes, it's doing exactly that :) though using state seems to be triggering too many rendering cycles that it's not scrolling properly :/